Content.Shared/ImmovableRod/ImmovableRodSystem.cs

```diff
@@ -0,0 +1,161 @@
+using Content.Shared.Body.Systems;
+using Content.Shared.Body.Components;
+using Content.Shared.Damage;
+using Content.Shared.Examine;
+using Content.Shared.Mobs.Components;
+using Content.Shared.Popups;
+using Content.Shared.Whitelist;
+using Robust.Shared.Audio.Systems;
+using Robust.Shared.Map;
+using Robust.Shared.Map.Components;
+using Robust.Shared.Physics.Components;
+using Robust.Shared.Physics.Events;
+using Robust.Shared.Physics.Systems;
+using Robust.Shared.Random;
+using Content.Shared.Movement.Systems;
+using Content.Shared.Damage.Components;
+using Content.Shared.Damage.Systems;
+
+namespace Content.Shared.ImmovableRod;
+
+public abstract class SharedImmovableRodSystem : EntitySystem
+{
+    [Dependency] private readonly IRobustRandom _random = default!;
+
+    [Dependency] private readonly SharedBodySystem _bodySystem = default!;
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+    [Dependency] private readonly SharedPhysicsSystem _physics = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly DamageableSystem _damageable = default!;
+    [Dependency] private readonly SharedTransformSystem _transform = default!;
+    [Dependency] private readonly SharedMapSystem _map = default!;
+    [Dependency] private readonly EntityWhitelistSystem _whitelist = default!;
+
+    [Dependency] private readonly StaminaSystem _stamina = default!;
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        // we are deliberately including paused entities. rod hungers for all
+        foreach (var (rod, trans) in EntityManager.EntityQuery<ImmovableRodComponent, TransformComponent>(true))
+        {
+            if (!rod.DestroyTiles)
+                continue;
+
+            if (!TryComp<MapGridComponent>(trans.GridUid, out var grid))
+                continue;
+
+            _map.SetTile(trans.GridUid.Value, grid, trans.Coordinates, Tile.Empty);
```

Comment:
should update this to `AllEntityQuery<T>` (this is the enumerator version) while you're here.